### ISSUE TITLE
Normalization and BatchNormalization layer does not rescale or normalize inputs, am I missing something?

### ISSUE THREAD

**Issue by jayagami** [keras-team/keras#16296]
I opened a same issue in tensorflow, but I found it could be keras probelm.

**System information**

colab with tf-2.8
[old-tf_env.txt](https://github.com/tensorflow/tensorflow/files/8330439/old-tf_env.txt)

---

**Describe the current behavior**

Normalization and Batch Normalization layer does not rescale or normalize inputs as excepted, the inputs barely not changing.

---

**Standalone code to reproduce the issue**

###  Normalization layer
```python
tfdata = np.array([[1,2,3,4,5]],dtype=float)
print("tfdata:" ,tfdata)
print("avg: ", np.average(tfdata))
print("var: ", np.var(tfdata))
```
>tfdata: [[1. 2. 3. 4. 5.]]
avg:  3.0
var:  2.0

```python
normalization_layer = tf.keras.layers.Normalization(axis=-1)
normalized_data = normalization_layer(tfdata)
print(normalized_data.numpy())
print("normalized avg: ", np.average(normalized_data.numpy()))
print("normalized var: ", np.var(normalized_data.numpy()))
```
**output:** 

>[[1. 2. 3. 4. 5.]]
normalized avg:  3.0
normalized var:  2.0

**This output is unchanged**

### BatchNormalization layer

```python

batch_normalization_layer = tf.keras.layers.BatchNormalization(axis=-1,scale=True,center=True)
batch_normalized_data = batch_normalization_layer(tfdata)
print(batch_normalized_data.numpy())
print("batch normalized avg: ", np.average(batch_normalized_data.numpy()))
print("batch normalized var: ", np.var(batch_normalized_data.numpy()))
```
**output:**
>[[0.9995004 1.9990008 2.9985013 3.9980016 4.997502 ]]
batch normalized avg:  2.9985013
batch normalized var:  1.9980018

---

**This output still unchanged!**

### LayerNorm

```python
tdata = torch.tensor([[1,2,3,4,5],[6,7,8,9,10]],dtype=torch.float32)
normalization_layer=torch.nn.BatchNorm1d(num_features=5)
normalized_data = normalization_layer(tdata).detach().numpy()
print(normalized_data)
print("normalized avg: ", np.average(normalized_data))
print("normalized var: ", np.var(normalized_data,))
```

**output**

>[[-0.9999992  -0.99999917 -0.9999991  -0.9999993  -0.9999992 ]
 [ 0.99999917  0.9999992   0.9999993   0.9999991   0.99999917]]
normalized avg:  1.7881394e-08
normalized var:  0.9999984

---

At first, I thought it could be different implementation, so I checked the code.

The call method is quite short, looks like mean and variance are never calculated.
I debugged this code and verified my guess.

https://github.com/keras-team/keras/blob/v2.8.0/keras/layers/preprocessing/normalization.py#L300-L306

```python
  def call(self, inputs):
    inputs = self._standardize_inputs(inputs)
    # The base layer automatically casts floating-point inputs, but we
    # explicitly cast here to also allow integer inputs to be passed
    inputs = tf.cast(inputs, self.compute_dtype)
    return ((inputs - self.mean) /
            tf.maximum(tf.sqrt(self.variance), backend.epsilon()))
```

https://github.com/keras-team/keras/blob/v2.8.0/keras/layers/preprocessing/normalization.py#L244-L298

**Comment by markub3327** [comment]
@jayagami 

When using `tf.keras.layers.Normalization` you must `.adapt()` before !!!

```python
normalization_layer = tf.keras.layers.Normalization(axis=-1)
layer.adapt(tfdata)
normalized_data = normalization_layer(tfdata)
print(normalized_data.numpy())
print("normalized avg: ", np.average(normalized_data.numpy()))
print("normalized var: ", np.var(normalized_data.numpy()))
```

**Comment by jayagami** [comment]
> @jayagami
> 
> When using `tf.keras.layers.Normalization` you must adapt it (`.adapt()`) before !!!
> 
> More info here: https://keras.io/api/layers/preprocessing_layers/numerical/normalization/
> 
> ```python
> normalization_layer = tf.keras.layers.Normalization(axis=-1)
> layer.adapt(tfdata)
> normalized_data = normalization_layer(tfdata)
> print(normalized_data.numpy())
> print("normalized avg: ", np.average(normalized_data.numpy()))
> print("normalized var: ", np.var(normalized_data.numpy()))
> ```
> 
> The BatchNormalization calculate mean and varience only in training mode !!! You must train it before using in prediction manner.
> 
> ```python
> batch_normalization_layer = tf.keras.layers.BatchNormalization(axis=-1,scale=True,center=True)
> batch_normalized_data = batch_normalization_layer(tfdata, training=True)
> print(batch_normalized_data.numpy())
> print("batch normalized avg: ", np.average(batch_normalized_data.numpy()))
> print("batch normalized var: ", np.var(batch_normalized_data.numpy()))
> 
> batch_normalized_data = batch_normalization_layer(tfdata, training=False)
> print(batch_normalized_data.numpy())
> print("batch normalized avg: ", np.average(batch_normalized_data.numpy()))
> print("batch normalized var: ", np.var(batch_normalized_data.numpy()))
> ```

Thanks for your explanation!
I re-checked the documentation and found it right in front of me, surprised I hadn't noticed this paragraph before.
I understand the need for this design, which is very different from torch. I should have paid more attention to these details.

**Comment by markub3327** [comment]
@fchollet  I solved it for Keras.
